ids/node_id.go

```diff
@@ -60,35 +52,60 @@ func (id *NodeID) UnmarshalJSON(b []byte) error {
 	}
 
 	var err error
-	*id, err = NodeIDFromString(str[1:lastIndex])
+	*nodeID, err = NodeIDFromString(str[1:lastIndex])
 	return err
 }
 
-func (id *NodeID) UnmarshalText(text []byte) error {
-	return id.UnmarshalJSON(text)
+func (nodeID NodeID) MarshalText() ([]byte, error) {
+	return []byte(nodeID.String()), nil
+}
+
+func (nodeID *NodeID) UnmarshalText(text []byte) error {
+	return nodeID.UnmarshalJSON(text)
```

Comment:
Shouldn't this be something along the lines of this? I don't think we should be unmarshalling json here:
```
nodeID, err := NewNodeIDFromBytes(text)
*nodeID = nodeID
return err
```

We should add a test for this